pkg/controller/upload-controller.go

```diff
@@ -370,7 +370,16 @@ func (c *UploadController) syncHandler(key string) error {
 		}
 
 		// delete pod
-		err = c.deletePod(pvc.Namespace, resourceName)
+		// we're using a req struct for now until we can normalize the controllers a bit more and share things like lister, client etc
+		// this way it's easy to stuff everything into an easy request struct, and can extend aditional behaviors if we want going forward
+		dReq := podDeleteRequest{
+			namespace: pvc.Namespace,
+			podName:   resourceName,
+			podLister: c.podLister,
+			k8sClient: c.client,
+		}
+
+		err = deletePod(dReq)
```

Comment:
This code path deletes the pod/service when the upload annotation is removed (maybe the user changed their mind about uploading).

To delete the pod when an upload succeeds, add something around here:  https://github.com/kubevirt/containerized-data-importer/blob/6327c899a2eb7d546ccf557a0578a4fcf664ef98/pkg/controller/upload-controller.go#L418

Reply:
Ahh!  I honestly didn't notice that was a user update on the annotation.  I'll take a look at adding another defer podDelete after success, thanks!